### ISSUE TITLE
Can not add both python and python3 support in OSX

### ISSUE THREAD

**Issue by magiclyx** [wsdjeg/SpaceVim#1251]
Spacevim requires python and python3 support，But vim compiled with both, or with 'dynamic' is only available in Windows versions. Mac and parts of *nix can only use one version of Python. 

This is a very early issue. I do not know if there is a solution now. 

**I tried to install vim with HomeBrew.** 
When I specify --with-python3 options, it removes python2 support.  I found an issue in HomeBrew page, The final solution is to remove python2 support when use --with-python3. This is consistent with what I actually tested.
https://github.com/Homebrew/legacy-homebrew/issues/20327

**I tried to compile vim myself**
I tried two kinds of compile parameters:

``` bash
--enable-pythoninterp=yes \
--enable-python3interp=yes \
```
or 
``` bash
--enable-pythoninterp=dynamic \
--with-python-config-dir=xxx
--enable-python3interp=dynamic \
 --with-python3-config-dir=xxx
```

"vim --version" show that it successful support for python2 and python3, But it does not work correctly when I run a python codes in vim scripts. SpaceVim also get a python error on launch:

``` python
Error detected while processing function rtags#InitPython:
line    7:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/yuxiliu/.cache/vimfiles/repos/github.com/lyuts/vim-rtags/plugin/vimrtags.py", line 4, in <module>
    import io
  File "/usr/local/Cellar/python/HEAD-e136981_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/io.py", line 51, in <module>
    import _io
ImportError: dlopen(/usr/local/Cellar/python/HEAD-e136981_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload/_io.so, 2): Symbol not found: __PyCodecInfo_GetIncrementalDecoder
  Referenced from: /usr/local/Cellar/python/HEAD-e136981_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload/_io.so
  Expected in: flat namespace
 in /usr/local/Cellar/python/HEAD-e136981_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload/_io.so
```

**I tried the same  compile parameters in a windows vm, it work correctly.**

**Comment by wsdjeg** [comment]
I do not use vim8， but as I know it should be able to support py/py3 at same time.

I recommend to use neovim. and it is easier to add py/py3/ruby support. BTW, it support lua by default.

**Comment by magiclyx** [comment]
Yes, I have tried, it can support py/py3 at the same time. But it just work correctly in windows. 

This is an old issue. it depends what the python package that is released on the operating system. This mailing list discussion about this:
https://www.mail-archive.com/vim_dev@googlegroups.com/msg22562.html


About of the OSX I used, python packaging doesn't link the python extensions against the relevant libpython shared library. This is what causes OSX vim packaging to require using RTLD_GLOBAL when using dlopen() to dynamically load the python language bindings. (Refer to my launching error message given in the issue).

Because of this issue, dynamic loading also disabled in OSX. So all the ways to enable both py2 and py3 are blocked. Unless you compile your own python and most of its dependent libraries. This is horrible！！

Most of the Linux/Unix system have the same Problem ! !  

**I suggest that this issue can be solved as soon as possible, after all, most vimuser is not working in windos and ubuntu。 They also do not use neovim because of plugins or bugs.**

I think it is easy to convert all python2 code to python3, or vice versa.
